### DISCUSSION TITLE
PEP 696 Support (Type Defaults for TypeVarsLike)

### DISCUSSION THREAD

**Discussion by Vezzp** [microsoft/pyright#4068] [General]
Are there any plans for supporting [this currently draft PEP](https://peps.python.org/pep-0696/) in pyright?

In [`typing_extensions` of version 4.4.0](https://pypi.org/project/typing-extensions/4.4.0/) they'd recently included this experimental feature.

Would be really nice to try it out with pyright!:)

**Comment by erictraut** [comment]
This feature is a bit of a heavy lift, so it will take a while to implement. There appears to be a [reference implementation](https://github.com/Gobot1234/mypy/tree/TypeVar-defaults) available on a branch of mypy if you want to try it out now.

**Reply by erictraut** [reply]
I've added an [enhancement request](https://github.com/microsoft/pyright/issues/4078) to track this work.
